src/secrets/actions/thunks.ts

```diff
@@ -0,0 +1,117 @@
+// Libraries
+import {normalize} from 'normalizr'
+
+// API
+import {
+  getOrgsSecrets as apiGetSecrets,
+  patchOrgsSecrets as apiUpdateSecret,
+  postOrgsSecretsDelete as apiDeleteSecret,
+} from 'src/client'
+
+// Schemas
+import {secretsSchema, arrayOfSecrets} from 'src/schemas/secrets'
+
+// Types
+import {Dispatch} from 'react'
+import {
+  RemoteDataState,
+  GetState,
+  ResourceType,
+  Secret,
+  SecretEntities,
+} from 'src/types'
+
+// Actions
+import {notify} from 'src/shared/actions/notifications'
+import {
+  getSecretsFailed,
+  upsertSecretFailed,
+  deleteSecretsFailed,
+} from 'src/shared/copy/notifications'
+import {
+  setSecrets,
+  setSecret,
+  removeSecret,
+  Action,
+} from 'src/secrets/actions/creators'
+
+// Utils
+import {getOrg} from 'src/organizations/selectors'
+import {getStatus} from 'src/resources/selectors'
+
+const makeEntitiesForSecrets = (response: Array<string>) => {
+  const processedData = []
+  response.forEach(element => processedData.push({id: element, key: element}))
+  return processedData
+}
+
+export const getSecrets = () => async (
+  dispatch: Dispatch<Action>,
+  getState: GetState
+) => {
+  const state = getState()
+  if (getStatus(state, ResourceType.Secrets) === RemoteDataState.NotStarted) {
+    dispatch(setSecrets(RemoteDataState.Loading))
+  }
+
+  const org = getOrg(state)
+
+  const resp = await apiGetSecrets({orgID: org.id})
+
+  if (resp.status === 200) {
+    const secrets = makeEntitiesForSecrets(resp.data.secrets)
+    const test = normalize<Secret, SecretEntities, string[]>(
+      secrets,
+      arrayOfSecrets
+    )
+    dispatch(setSecrets(RemoteDataState.Done, test))
+  } else {
+    console.error(resp.data.message)
+    dispatch(setSecrets(RemoteDataState.Error))
+    dispatch(notify(getSecretsFailed()))
+  }
+}
+
+export const upsertSecret = (_secretKey: string, secretValue: string) => async (
```

Comment:
why are we passing in the secretKey if we're never using it?

Reply:
It's used on line 81

Reply:
<img width="492" alt="Screen Shot 2021-06-17 at 11 51 55 AM" src="https://user-images.githubusercontent.com/19984220/122456458-71ec7a80-cf62-11eb-9a99-22fa98134308.png">

Not sure which line 81 you're referring to

Reply:
![image](https://user-images.githubusercontent.com/22055730/122457056-3eb6e500-cf7c-11eb-819a-eb85b4aafd38.png)
The one :point_down:

Reply:
That's a property that you're assigning a value to. 

> ![image](https://user-images.githubusercontent.com/22055730/122457056-3eb6e500-cf7c-11eb-819a-eb85b4aafd38.png)
> The one 👇

Reply:
If you remove the `_` from secretKey it'll throw a linter error because `_secretKey` is unused.

Reply:
![Uploading Screen Shot 2021-06-17 at 12.00.49 PM.png…]()

Reply:
<img width="270" alt="Screen Shot 2021-06-17 at 12 03 12 PM" src="https://user-images.githubusercontent.com/19984220/122457881-01def400-cf64-11eb-9f1e-59076e6a3877.png">

Reply:
Ah I see!  Added those brackets and will test this more heavily when the add button is implemented.  This probably saved me a bunch of confusion later - thanks!